src/keywords.rs

```diff
@@ -1,170 +1,193 @@
-        | terminated!(tag_no_case!("THEN"), keyword_follow_char)
-        | terminated!(tag_no_case!("TO"), keyword_follow_char)
-        | terminated!(tag_no_case!("TRANSACTION"), keyword_follow_char)
-        | terminated!(tag_no_case!("TRIGGER"), keyword_follow_char)
-        | terminated!(tag_no_case!("UNION"), keyword_follow_char)
-        | terminated!(tag_no_case!("UNIQUE"), keyword_follow_char)
-        | terminated!(tag_no_case!("UPDATE"), keyword_follow_char)
-        | terminated!(tag_no_case!("USING"), keyword_follow_char)
-        | terminated!(tag_no_case!("VACUUM"), keyword_follow_char)
-        | terminated!(tag_no_case!("VALUES"), keyword_follow_char)
-        | terminated!(tag_no_case!("VIEW"), keyword_follow_char)
-        | terminated!(tag_no_case!("VIRTUAL"), keyword_follow_char)
-        | terminated!(tag_no_case!("WHEN"), keyword_follow_char)
-        | terminated!(tag_no_case!("WHERE"), keyword_follow_char)
-        | terminated!(tag_no_case!("WITH"), keyword_follow_char)
-        | terminated!(tag_no_case!("WITHOUT"), keyword_follow_char)
-    )
-);
+fn keyword_s_to_z(i: &[u8]) -> IResult<&[u8], &[u8]> {
+    alt((
+        terminated(tag_no_case("SET"), keyword_follow_char),
+        terminated(tag_no_case("TABLE"), keyword_follow_char),
+        terminated(tag_no_case("TEMP"), keyword_follow_char),
+        terminated(tag_no_case("TEMPORARY"), keyword_follow_char),
+        terminated(tag_no_case("THEN"), keyword_follow_char),
+        terminated(tag_no_case("TO"), keyword_follow_char),
+        terminated(tag_no_case("TRANSACTION"), keyword_follow_char),
+        terminated(tag_no_case("TRIGGER"), keyword_follow_char),
+        terminated(tag_no_case("UNION"), keyword_follow_char),
+        terminated(tag_no_case("UNIQUE"), keyword_follow_char),
+        terminated(tag_no_case("UPDATE"), keyword_follow_char),
+        terminated(tag_no_case("USING"), keyword_follow_char),
+        terminated(tag_no_case("VACUUM"), keyword_follow_char),
+        terminated(tag_no_case("VALUES"), keyword_follow_char),
+        terminated(tag_no_case("VIEW"), keyword_follow_char),
+        terminated(tag_no_case("VIRTUAL"), keyword_follow_char),
+        terminated(tag_no_case("WHEN"), keyword_follow_char),
+        terminated(tag_no_case("WHERE"), keyword_follow_char),
+        terminated(tag_no_case("WITH"), keyword_follow_char),
+        terminated(tag_no_case("WITHOUT"), keyword_follow_char),
+    ))(i)
+}
 
 // Matches any SQL reserved keyword
-named!(pub sql_keyword<CompleteByteSlice, CompleteByteSlice>,
-    do_parse!(
-        kw: alt!(
-              keyword_a_to_c
-            | keyword_d_to_i
-            | keyword_j_to_s
-            | keyword_t_to_z) >>
-        (kw)
-    )
-);
+pub fn sql_keyword(i: &[u8]) -> IResult<&[u8], &[u8]> {
+    alt((
+        keyword_a_to_c,
+        keyword_c_to_e,
+        keyword_e_to_i,
+        keyword_i_to_o,
+        keyword_o_to_s,
+        keyword_s_to_z,
```

Comment:
I like how you moved the trailing identifier for the keyword functions to an open interval, it makes it easier to see that the alts overlap and contain the whole space.